### PR TITLE
Fix stat_type distrib on FluxPointsDataset if error or ul are not finite

### PR DESCRIPTION
**Pull request by QRemy**
Fix https://github.com/gammapy/gammapy/issues/5531#issuecomment-2447768585

### CODE REVIEW

File: examples/tutorials/analysis-1d/spectral_analysis.py [hunk 1]
```diff
@@ -500,11 +500,13 @@
 # --------------------
 #
 # Different statistic are available for the FluxPointDataset :
```
Comment (Astro-Kirsty) [inline examples/tutorials/analysis-1d/spectral_analysis.py:502]:
```suggestion
# Different statistic are available for the `~gammapy.datasets.FluxPointsDataset` :
```

**Comment by Astro-Kirsty** [comment]
Thanks @QRemy. On line 555 can you add a #, so that it renders correctly?
image

**Comment by QRemy** [comment]
@Astro-Kirsty Done.

**Review by Astro-Kirsty — APPROVED** [review]

**Review by registerrier — COMMENTED** [review]
Thanks @QRemy ! 

Could you just summarize in a few words what was wrong? Shouldn't this require a more complete test?

File: gammapy/datasets/flux_points.py [hunk 2]
```diff
@@ -440,20 +440,24 @@ def _stat_array_distrib(self):
...
+            mask_invalid = np.isnan(scale)
+            scale[mask_invalid] = self.data.dnde_err.data[mask_valid][mask_invalid]
         except AttributeError:
             scale = self.data.dnde_err.data[mask_valid]
-        stat[mask_valid] = -2 * np.log(
-            norm.pdf(value, loc=loc, scale=scale) / norm.pdf(loc, loc=loc, scale=scale)
-        )
 
-        mask_ul = self.data.is_ul.data & ~np.isnan(self.data.dnde_ul.data)
+        stat[mask_valid] = ((value - loc) / scale) ** 2
```
[the remark this answers is not in the document]
Comment (QRemy) [reply gammapy/datasets/flux_points.py:449]: changed to be consitant with https://github.com/gammapy/gammapy/blob/a65d9452cea9acd8343c99201598274b67140f03/gammapy/estimators/utils.py#L1201

File: gammapy/datasets/flux_points.py [hunk 3]
```diff
@@ -440,20 +440,24 @@ def _stat_array_distrib(self):
...
+
+        mask_ul = self.data.is_ul.data
         value = model[mask_ul]
         loc_ul = self.data.dnde_ul.data[mask_ul]
         scale_ul = self.data.dnde_ul.data[mask_ul]
-        stat[mask_ul] = 2 * np.log(
-            (erfc((loc_ul - value) / scale_ul) / 2)
-            / (erfc((loc_ul - 0) / scale_ul) / 2)
+        stat[mask_ul] = -2 * np.log(
```
[the remark this answers is not in the document]
Comment (QRemy) [reply gammapy/datasets/flux_points.py:455]: I changed the signs to be consistent with https://github.com/gammapy/gammapy/blob/a65d9452cea9acd8343c99201598274b67140f03/gammapy/estimators/utils.py#L1231-L1234

[the remark this answers is not in the document]
Comment (QRemy) [reply gammapy/datasets/flux_points.py:455]: now I'm not sure which one was wrong

File: gammapy/datasets/tests/test_flux_points.py [hunk 4]
```diff
@@ -112,7 +112,7 @@ def test_flux_point_dataset_stat(dataset):
     dataset.stat_type = "distrib"
     fit = Fit()
     fit.run([dataset])
-    assert_allclose(dataset.stat_sum(), 36.153428, rtol=1e-3)
+    assert_allclose(dataset.stat_sum(), 27.244609, rtol=1e-3)
```
Comment (QRemy) [inline gammapy/datasets/tests/test_flux_points.py:115]:
```suggestion
    assert_allclose(dataset.stat_sum(), 36.153428, rtol=1e-3)
```

File: gammapy/datasets/flux_points.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (QRemy) [reply gammapy/datasets/flux_points.py]: removed this change, the original sign match the one described in the paper (the other method sign seems also correct given how the TS is defined in the other case)

File: gammapy/datasets/tests/test_flux_points.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (QRemy) [reply gammapy/datasets/tests/test_flux_points.py]: removed that was due to the unecessary change of sig.

**Review by registerrier — APPROVED** [review]
Thanks @QRemy . Looks good to me.

**Pull request merged**